app/services/wrappers/base.rb

```diff
@@ -13,19 +13,61 @@ def client
     @client ||= Faraday.new(url: base_url, headers: default_headers)
   end
 
-  def get(path, params = {})
-    JSON.parse(client.get(path, params).env.response_body)
+  def handle_response(response)
+    if response.success?
+      return(response.body.is_a?(Hash) ? response.body : JSON.parse(response.body))
+    else
+      handle_error(response)
+    end
   end
 
-  def post(path, payload = {})
-    JSON.parse(client.post(path, payload.to_json).env.response_body)
+  def get(path, params = {}, skip_handle_response = false)
+    response = client.get(path, params)
+    skip_handle_response ? response : handle_response(response)
+  rescue Faraday::Error => e
+    handle_faraday_error(e)
   end
 
-  def put(path, payload = {})
-    JSON.parse(client.put(path, payload.to_json).env.response_body)
+  def post(path, payload = {}, skip_handle_response = false)
+    response = client.post(path, payload)
+    skip_handle_response ? response : handle_response(response)
+  rescue Faraday::Error => e
+    handle_faraday_error(e)
   end
 
-  def delete(path)
-    JSON.parse(client.delete(path).env.response_body)
+  def put(path, payload = {}, skip_handle_response = false)
+    response = client.put(path, payload)
+    skip_handle_response ? response : handle_response(response)
```

Comment:
I see this ternary several times in this file. Might be a bit cleaner rename `skip_handle_response` to some other property like `skip_response_processing` and handle this flag inside the `handle_response`. This way you just have perform this logic at one place